src/app/payment/paymentMethod/BarclaycardPaymentMethod.spec.tsx

```diff
@@ -1,92 +1,28 @@
 import { mount, ReactWrapper } from 'enzyme';
-import { Formik } from 'formik';
-import { noop } from 'lodash';
-import React, { FunctionComponent } from 'react';
-
-import { CheckoutProvider } from '../../checkout';
-import { getStoreConfig } from '../../config/config.mock';
-import { createLocaleContext, LocaleContext, LocaleContextType } from '../../locale';
-import { getPaymentMethod } from '../payment-methods.mock';
+import React from 'react';
 
+import BarclaycardPaymentMethod from './BarclaycardPaymentMethod';
 import HostedWidgetPaymentMethod, { HostedWidgetPaymentMethodProps } from './HostedWidgetPaymentMethod';
-import { default as PaymentMethodComponent, PaymentMethodProps } from './PaymentMethod';
-import PaymentMethodId from './PaymentMethodId';
+import { PaymentMethodProps } from './PaymentMethod';
 
 describe('when using Barclaycard payment', () => {
-    let method: PaymentMethod;
-    let checkoutService: CheckoutService;
-    let checkoutState: CheckoutSelectors;
-    let defaultProps: PaymentMethodProps;
-    let localeContext: LocaleContextType;
-    let PaymentMethodTest: FunctionComponent<PaymentMethodProps>;
-
-    beforeEach(() => {
-        defaultProps = {
-            method: getPaymentMethod(),
-            onUnhandledError: jest.fn(),
-        };
-
-        checkoutService = createCheckoutService();
-        checkoutState = checkoutService.getState();
-        localeContext = createLocaleContext(getStoreConfig());
-        method = { ...getPaymentMethod(), id: PaymentMethodId.Amazon };
-
-        jest.spyOn(checkoutState.data, 'getConfig')
-            .mockReturnValue(getStoreConfig());
-
-        jest.spyOn(checkoutService, 'deinitializePayment')
-            .mockResolvedValue(checkoutState);
-
-        jest.spyOn(checkoutService, 'initializePayment')
-            .mockResolvedValue(checkoutState);
-
-        PaymentMethodTest = props => (
-            <CheckoutProvider checkoutService={ checkoutService }>
-                <LocaleContext.Provider value={ localeContext }>
-                    <Formik
-                        initialValues={ {} }
-                        onSubmit={ noop }
-                    >
-                        <PaymentMethodComponent { ...props } />
-                    </Formik>
-                </LocaleContext.Provider>
-            </CheckoutProvider>
-        );
-    });
-
-    it('renders as hosted widget method', () => {
-        const container = mount(<PaymentMethodTest { ...defaultProps } method={ method } />);
+    const defaultProps: PaymentMethodProps = {
+        method: {
+            id: 'barclaycard',
+            method: 'barclaycard',
+            supportedCards: [],
+            config: {},
+            type: 'card',
+            gateway: 'barclaycard',
+        },
+        isEmbedded: true,
+    };
+
+    it('should mount Barclaycard', () => {
+        const container = mount(<BarclaycardPaymentMethod { ...defaultProps } />);
```

Comment:
I'm seeing TS errors here:

`Type '{ method: any; isEmbedded?: boolean | undefined; isUsingMultiShipping?: boolean | undefined; onUnhandledError?(error: Error): void; submitForm?(): void; }' is missing the following properties from type 'BarclaycardPaymentMethodProps': deinitializePayment, initializePayment`

Which is because we're trying to apply `PaymentMethodProps` to `BarclaycardPaymentMethod` - which is not an exact fit.

We can forget about `PaymentMethodProps` in this context - that component and its props are of a higher scope that this test is agnostic to.

So default props can just be:

```tsx
const defaultProps = {
    method: {
        id: 'barclaycard',
        method: 'barclaycard',
        supportedCards: [],
        config: {},
        type: 'card',
        gateway: 'barclaycard',
    },
    deinitializePayment: jest.fn(),
    initializePayment: jest.fn(),
    isEmbedded: true,
};
```